applications/RomApplication/custom_strategies/rom_builder_and_solver.h

```diff
@@ -411,25 +442,26 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
 
                 if (element_is_active){
                     //calculate elemental contribution
-                    pScheme->CalculateSystemContributions(*(it_el.base()), LHS_Contribution, RHS_Contribution, EquationId, CurrentProcessInfo);
+                    pScheme->CalculateSystemContributions(*it_el, LHS_Contribution, RHS_Contribution, EquationId, CurrentProcessInfo);
                     Element::DofsVectorType dofs;
                     it_el->GetDofList(dofs, CurrentProcessInfo);
                     const auto &geom = it_el->GetGeometry();
                     if(PhiElemental.size1() != dofs.size() || PhiElemental.size2() != mRomDofs)
                         PhiElemental.resize(dofs.size(), mRomDofs,false);
                     GetPhiElemental(PhiElemental, dofs, geom);
                     Matrix aux = prod(LHS_Contribution, PhiElemental);

```

Comment:
only an optimization thing, however it would be better to allocate aux once outside of the loop, resize it the first time (by the size of the space) and make it threadprivate. Then just overwrite it on each thread 

the point is that this would allow to avoid creating a new, rather big matrix, for every element (remember that memory allocation is one of the most expensive operations on modern computers!)